ubg/src/search/Search.js

```diff
@@ -0,0 +1,133 @@
+import React, {useEffect} from 'react';
+import {makeStyles} from '@material-ui/core/styles';
+import Box from '@material-ui/core/Box';
+import Grid from '@material-ui/core/Grid';
+import Button from '@material-ui/core/Button';
+import Navbar from '../common/Navbar';
+import GameCard from './GameCard';
+import Filter from './Filter';
+import {useFirestore} from 'reactfire';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1,
+  },
+  button: {
+    marginTop: theme.spacing(1.5),
+    marginRight: theme.spacing(3),
+    float: 'right',
+  },
+}));
+
+let initialize = false;
+
+/**
+ * @return {ReactElement} Search page with filter and search result
+ */
+export default function Search() {
+  const classes = useStyles();
+  const ref = useFirestore().collection('games');
+  const [minAge, setMinAge] = React.useState(21);
+  const [minPlayer, setMinPlayer] = React.useState(1);
+  const [maxPlayer, setMaxPlayer] = React.useState('8+');
+  const [minTime, setMinTime] = React.useState(5);
+  const [maxTime, setMaxTime] = React.useState('240+');
+  const [games, setGames] = React.useState([]);
+  const handleFilter = () => {
+    const newGames = [];
+    let maxP = maxPlayer;
+    if (typeof maxP === 'string') {
+      maxP = Number.MAX_SAFE_INTEGER;
+    }
+    let maxT = maxTime;
+    if (typeof maxT === 'string') {
+      maxT = Number.MAX_SAFE_INTEGER;
+    }
+    if (minPlayer > maxP) {
+      setMaxPlayer(minPlayer);
+      maxP = minPlayer;
+    }
+    if (minTime > maxT) {
+      setMaxTime(minTime);
+      maxT = minTime;
+    }
+    ref.orderBy('rating', 'desc')
+        .get()
+        .then(function(querySnapshot) {
+          querySnapshot.forEach(function(doc) {
+            if (doc.data()['minPlayer'] <= maxP &&
+            minPlayer <= doc.data()['maxPlayer'] &&
+            doc.data()['minPlaytime'] <= maxT &&
+            minTime <= doc.data()['maxPlaytime'] &&
+            doc.data()['minAge'] <= minAge) {
+              newGames.push(doc.data());
+            }
+          });
+          setGames(newGames);
+        })
+        .catch(function(error) {
+          console.log('Error getting documents: ', error);
+        });
+  };
+  useEffect(() => {
+    // using a hack to make useEffect act as onLoad()
+    if (initialize === false) {
+    // console.log("4");
+      const newGames = [];
+      ref.orderBy('rating', 'desc')
+          .get()
+          .then(function(querySnapshot) {
+            querySnapshot.forEach(function(doc) {
+              newGames.push(doc.data());
+            });
+            setGames(newGames);
+            initialize = true;
+          })
+          .catch(function(error) {
+            console.log('Error getting documents: ', error);
+          });
+    }
+  }, [ref]);
+
+  return (
+    <div>
+      <Navbar/>
+      <Box boxShadow={1} m={10}>
+        <Filter label = "Minimum Age" value={minAge} menu={[8, 10, 14, 16, 21]}
```

Comment:
Same with all of the components on this file, I would put every prop in a different line.